java/algorithm/src/test/java/org/apache/arrow/algorithm/search/TestParallelSearcher.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.search;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test cases for {@link ParallelSearcher}.
+ */
+public class TestParallelSearcher {
```

Comment:
Could you add some benchmarks too? It would be helpful to have a way to track performance.

Reply:
Good suggestion. Thank you.
I have added a benchmark for different numbers of threads. Below are the current performance data. Note that there are regressions in the data. This is because the performance is highly dependent on the position of the matched key. 

Benchmark                                   (numThreads)  Mode  Cnt     Score      Error  Units
ParallelSearcherBenchmarks.searchBenchmark             1  avgt    5  7847.050 ± 1608.563  us/op
ParallelSearcherBenchmarks.searchBenchmark             2  avgt    5  7793.512 ±   32.010  us/op
ParallelSearcherBenchmarks.searchBenchmark             5  avgt    5  3143.602 ±   49.519  us/op
ParallelSearcherBenchmarks.searchBenchmark            10  avgt    5   826.998 ±   18.675  us/op
ParallelSearcherBenchmarks.searchBenchmark            20  avgt    5  1605.172 ±   73.040  us/op
ParallelSearcherBenchmarks.searchBenchmark            50  avgt    5   726.294 ±   10.599  us/op
ParallelSearcherBenchmarks.searchBenchmark           100  avgt    5   585.456 ±   82.108  us/op

Reply:
Thanks!